clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1281,17 +1259,11 @@ private void recordTopicFetchMetrics(String topic, int bytes, int records) {
                 Map<String, String> metricTags = Collections.singletonMap("topic", topic.replace('.', '_'));
 
                 bytesFetched = this.metrics.sensor(name);
-                bytesFetched.add(this.metrics.metricName("fetch-size-avg",
-                        this.metricGrpName,
-                        "The average number of bytes fetched per request for topic " + topic,
+                bytesFetched.add(this.metrics.metricInstance(metricsRegistry.topicFetchSizeAvg,
```

Comment:
I do think the MetricNameTemplate class is necessary (I described it up above).

But I suppose the external FetcherMetricsRegistry class isn't entirely necessary. I suppose they could have been simply placed inside this class.

I thought that having them external would make it easier for the ConsumerMetrics class to pull them in. In a future checkin, there are going to be other *MetricsRegistry.java classes, like SenderMetricsRegistry and ConsumerCoordinatorMetricsRegistry.java, and it seemed cleaner to have all the metrics in centralized classes, according to functionality.